internal/verifier/pprof.go

```diff
@@ -0,0 +1,45 @@
+package verifier
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime/pprof"
+	"time"
+)
+
+func (verifier *Verifier) MaybeStartPeriodicHeapProfileCollection(ctx context.Context) {
+	if verifier.pprofInterval == 0 {
+		return
+	}
+
+	go func() {
+		ticker := time.NewTicker(verifier.pprofInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				collectHeapUsage()
+			}
+		}
+	}()
+
+}
+
+func collectHeapUsage() {
+	heapFileName := fmt.Sprintf("heap-%s.out", time.Now().UTC().Format("20060102T150405Z"))
+	heapFile, err := os.Create(heapFileName)
+	defer heapFile.Close()
+
+	if err != nil {
+		panic(err)
+	}
+
+	err = pprof.Lookup("heap").WriteTo(heapFile, 0)
+	if err != nil {
+		heapFile.WriteString(err.Error())
```

Comment:
Why write the error to the file rather than panicking?

Reply:
I was thinking about logging and skipping the errors. But it's not intended to be used for production, panicking should be fine.